### PR TITLE
chore: cargo updates

### PR DESCRIPTION
**Pull request by tiagolobocastro**
chore: update etcd-client to latest

This brings on the new tonic which exposes its Streams as non-Sync and so we had to change some
methods to &mut, which is not great but does resolve this huddle for now without having to
add some sort of trampoline like channels.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore: upgrade to latest dependencies submodule
This trims out the dependencies as we're trying to upgrade tonic/prost.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore: upgrade to snafu 0.7.1

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore: upgrade tonic and prost crates

The newest tonic has a simpler Router trait, which might allow us to tidy up the
common service builder library, maybe..

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

fix(nix-shell): add modules required for commitlint

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore: update to latest mayastor-api submodule

todo: use the new crate rather than keeping local code.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

File: control-plane/grpc/src/context.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ impl<C: Clone> Client<C> {
     {
         let context = Context::new(options);
         let endpoint = context.endpoint(uri);
-        let channel = endpoint.connect_lazy().unwrap();
+        let channel = endpoint.connect_lazy();
```
Comment (Abhinandan-Purkait) [inline control-plane/grpc/src/context.rs:164]: Question: Did the function signature change? Is there any change in behaviour?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/grpc/src/context.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/grpc/src/context.rs:164]: It did yes, but no change in behaviour, it never used to return any error anyway.

Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/context.rs:164]: Okay

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**